packages/Laravel/src/Rector/Class_/InlineValidationRulesToArrayDefinitionRector.php

```diff
@@ -0,0 +1,184 @@
+        return new RectorDefinition('Transforms inline validation rules to array definition', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Illuminate\Foundation\Http\FormRequest;
+
+class SomeClass extends FormRequest
+{
+    public function rules(): array
+    {
+        return [
+            'someAttribute' => 'required|string|exists:' . SomeModel::class . 'id',
+        ];
+    }
+}
+CODE_SAMPLE
+,
+                <<<'CODE_SAMPLE'
+use Illuminate\Foundation\Http\FormRequest;
+
+class SomeClass extends FormRequest
+{
+    public function rules(): array
+    {
+        return [
+            'someAttribute' => ['required', 'string', \Illuminate\Validation\Rule::exists(SomeModel::class, 'id')],
+        ];
+    }
+}
+CODE_SAMPLE
+
+            )
+        ]);
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Node\Stmt\Class_::class];
+    }
+
+    /**
+     * @param Node\Stmt\Class_ $node
+     * @return Node|null
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $className = $this->getName($node);
+        if ($className === null || !is_subclass_of($className, 'Illuminate\Foundation\Http\FormRequest', true)) {
+            return null;
+        }
+
+        $rules = $node->getMethod('rules');
+        if ($rules === null || $rules->stmts === null) {
+            return null;
+        }
+
+        foreach ($rules->stmts as $statement) {
+            if (!$statement instanceof Node\Stmt\Return_) {
+                continue;
+            }
+
+            $expression = $statement->expr;
+            if (!$expression instanceof Node\Expr\Array_) {
+                continue;
+            }
+
+            foreach ($expression->items as $item) {
+                if (!$item->value instanceof Node\Scalar\String_ && !$item->value instanceof Node\Expr\BinaryOp\Concat) {
+                    continue;
+                }
+
+                $newRules = $this->transformRulesSet($item->value);
+                foreach ($newRules as &$newRule) {
+                    if (!($newRule instanceof Node\Expr\BinaryOp\Concat)) {
+                        continue;
+                    }
+
+                    try {
+                        $fullString = $this->transformConcatExpression($newRule);
+                    } catch (\LogicException $exception) {
+                        continue;
+                    }
+
+                    $matchesExist = preg_match('/^exists:(\w+),(\w+)$/', $fullString, $matches);
+                    if ($matchesExist === false || $matchesExist === 0) {
+                        continue;
+                    }
+
+                    $newRule = new Node\Expr\StaticCall(
+                        new Node\Name('\Illuminate\Validation\Rule'),
+                        'exists',
+                        [
+                            new Node\Arg(
+                                new Node\Expr\ClassConstFetch(
+                                    new Node\Name($matches[1]),
```

Comment:
Instead of `$matches[1]` better use named variable. Now I have no ides what is it